src/overlays/actors/ovl_En_Az/z_en_az.c

```diff
@@ -39,108 +83,1755 @@ const ActorInit En_Az_InitVars = {
+            break;
+        case 3:
+            phi_v1 = (gSaveContext.save.entranceIndex != 0x8E10) || (gSaveContext.save.weekEventReg[24] & 4);
+            break;
+        case 5:
+            phi_v1 = gSaveContext.save.entranceIndex != 0x8E20;
+            break;
+        case 6:
+            phi_v1 = (gSaveContext.save.entranceIndex == 0x8E00) && !(gSaveContext.save.weekEventReg[93] & 1);
+            phi_v1 = !phi_v1;
+            break;
+        default:
+            phi_v1 = true;
+            break;
+    }
+    if (phi_v1) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    this->unk_2F8 = D_80A9915C[GET_PARAM1(thisx)];
+    if (GET_PARAM1(thisx) >= 0) {
+        sp4E = D_80A9914C[GET_PARAM1(thisx)];
+    } else {
+        sp4E = -1;
+    }
+    if (this->unk_2F8 == 0) {
+        this->unk_374 |= 2;
+    }
+    SubS_FillCutscenesList(&this->actor, &this->unk_3D0, 1);
+    if (D_80A9913C == NULL) {
+        D_80A9913C = THIS;
+        this->unk_374 |= 1;
+    }
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 30.0f);
+    if (this->unk_374 & 2) {
+        SkelAnime_InitFlex(play, &this->skelAnime, &object_az_Skel_007438, &object_az_Anim_00C94C, this->jointTable,
+                           this->morphTable, 0x18);
+        Actor_SetScale(&this->actor, 0.012f);
+    } else {
+        SkelAnime_InitFlex(play, &this->skelAnime, &object_az_Skel_017990, &object_az_Anim_00C94C, this->jointTable,
+                           this->morphTable, 0x18);
+    }
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    if (this->unk_374 & 2) {
+        this->collider.dim.radius = this->collider.dim.radius * 1.2f;
+        this->collider.dim.height = this->collider.dim.height * 1.2f;
+        this->collider.dim.yShift = this->collider.dim.yShift * 1.2f;
+    }
+    Actor_UpdateBgCheckInfo(play, &this->actor, 0.0f, 0.0f, 0.0f, 5);
+    if ((this->actor.bgCheckFlags & 0x20) && (this->actor.depthInWater > 22.0f)) {
+        this->unk_374 |= 0x100;
+        this->unk_376 |= 0x100;
+    }
+    Animation_Change(&this->skelAnime, D_80A99010[0].animation, 1.0f,
+                     Animation_GetLastFrame(&D_80A99010[0].animation->common) * Rand_ZeroOne(),
+                     Animation_GetLastFrame(&D_80A99010[0].animation->common), D_80A99010[0].mode,
+                     D_80A99010[0].morphFrames);
+    this->unk_37E = 0;
+    this->unk_380 = 0;
+    this->unk_384 = 0;
+    this->actor.gravity = -1.0f;
+    this->unk_376 = this->unk_374;
+    SubS_ChangeAnimationBySpeedInfo(&this->skelAnime, D_80A99010, 0, &this->animIndex);
+    this->skelAnime.curFrame = Rand_ZeroOne() * this->skelAnime.endFrame;
+
+    switch (gSaveContext.save.entranceIndex) {
+        case 0x8E00:
+            if (gSaveContext.save.weekEventReg[93] & 1) {
+                this->unk_2FA = 5;
+                if (this->unk_374 & 2) {
+                    this->actor.flags |= 9;
```

Comment:
```suggestion
                    this->actor.flags |= (ACTOR_FLAG_1 | ACTOR_FLAG_8);
```
Could you fill in all the actor flag macros? There's a lot in this file so I won't comment on any more